File: plateform/frontend/src/types/agentAudit.ts
export interface AgentAuditFilters {
  agent_code: string;
  status: string;
  decision: string;
  incident: string;
  detection: string;
  device: string;
  search: string;
}

export interface AgentCount {
  key: string;
  count: number;
}

export interface AgentAuditSummary {
  total_runs: number;
  successful_runs: number;
  failed_runs: number;
  blocked_actions: number;
  average_duration_ms: number | null;
  runs_by_agent: AgentCount[];
  stages_reached: AgentCount[];
  last_run_at: string | null;
  unmapped_identity_count: number;
  data_mode: string;
  reference_time: string;
  note: string;
}

export interface AgentAuditEventSummary {
  id: string;
  public_id: string;
  agent_run_id: string | null;
  agent_code: string;
  contract_version: string;
  pipeline_stage: string;
  node_name: string | null;
  sequence_number: number;
  event_type: string;
  status: string;
  summary: string;
  incident_public_id: string | null;
  detection_public_id: string | null;
  asset_public_id: string | null;
  external_cluster_id: string | null;
  external_device_id: string | null;
  error_code: string | null;
  error_message: string | null;
  duration_ms: number | null;
  data_mode: string;
  occurred_at: string;
  reasoning_summary: string | null;
}

export interface AgentAuditEventDetail extends AgentAuditEventSummary {
  input_summary: Record<string, unknown>;
  output_summary: Record<string, unknown>;
  reasoning_trace: unknown;
  safety_checks: unknown;
}

export interface LinkedFinding {
  classification: string;
  severity_tier: number;
  confidence_score: number;
  operator_justification: string | null;
  mapping_status: string;
  external_cluster_id: string | null;
  mapped_detection_id: string | null;
}

export interface AgentAuditRunSummary {
  id: string;
  public_id: string;
  agent_code: string;
  agent_type: string;
  contract_version: string;
  pipeline_stages: string[];
  source_type: string;
  source_public_id: string | null;
  status: string;
  classification: string | null;
  investigation_severity: number | null;
  decision: string | null;
  duration_ms: number | null;
  started_at: string;
  related_incident: string | null;
  related_detection: string | null;
  related_device: string | null;
  external_cluster_id: string | null;
  data_mode: string;
  blocked: boolean;
  contract_unconfirmed: boolean;
  mapping_status: string | null;
}

export interface AgentAuditRunDetail extends AgentAuditRunSummary {
  events: AgentAuditEventDetail[];
  findings: LinkedFinding[];
  recommendation_decisions: string[];
  valve_command_sent: boolean;
  valve_command_confirmed: boolean;
  notification_sent: boolean;
  safety_status: string | null;
  note: string;
}

export interface AgentAuditRunListResponse {
  items: AgentAuditRunSummary[];
  total: number;
  page: number;
  page_size: number;
  data_mode: string;
  reference_time: string;
}

File: plateform/backend/app/schemas/agent_audit.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field

class AgentCount(BaseModel):
    key: str
    count: int


class AgentAuditEventSummary(BaseModel):
    id: str
    public_id: str
    agent_run_id: str | None
    agent_code: str
    contract_version: str
    pipeline_stage: str
    node_name: str | None
    sequence_number: int
    event_type: str
    status: str
    summary: str
    incident_public_id: str | None
    detection_public_id: str | None
    asset_public_id: str | None
    external_cluster_id: str | None
    external_device_id: str | None
    error_code: str | None
    error_message: str | None
    duration_ms: int | None
    data_mode: str = "ingested"
    occurred_at: datetime
    reasoning_summary: str | None


class AgentAuditEventDetail(AgentAuditEventSummary):
    input_summary: dict[str, Any]
    output_summary: dict[str, Any]
    reasoning_trace: Any
    safety_checks: Any


class LinkedFinding(BaseModel):
    classification: str
    severity_tier: int
    confidence_score: float
    operator_justification: str | None
    mapping_status: str
    external_cluster_id: str | None
    mapped_detection_id: str | None


class AgentAuditRunSummary(BaseModel):
    id: str
    public_id: str
    agent_code: str
    agent_type: str
    contract_version: str
    pipeline_stages: list[str] = Field(default_factory=list)
    source_type: str
    source_public_id: str | None
    status: str
    classification: str | None
    investigation_severity: int | None
    decision: str | None
    duration_ms: int | None
    started_at: datetime
    related_incident: str | None
    related_detection: str | None
    related_device: str | None
    external_cluster_id: str | None
    data_mode: str
    blocked: bool
    contract_unconfirmed: bool
    mapping_status: str | None


class AgentAuditRunDetail(AgentAuditRunSummary):
    events: list[AgentAuditEventDetail]
    findings: list[LinkedFinding]
    recommendation_decisions: list[str]
    valve_command_sent: bool
    valve_command_confirmed: bool
    notification_sent: bool
    safety_status: str | None
    note: str


class AgentAuditRunListResponse(BaseModel):
    items: list[AgentAuditRunSummary]
    total: int
    page: int
    page_size: int
    data_mode: str = "ingested"
    reference_time: datetime


class AgentAuditEventListResponse(BaseModel):
    items: list[AgentAuditEventSummary]
    total: int
    page: int
    page_size: int
    data_mode: str = "ingested"
    reference_time: datetime


class AgentAuditSummary(BaseModel):
    total_runs: int
    successful_runs: int
    failed_runs: int
    blocked_actions: int
    average_duration_ms: float | None
    runs_by_agent: list[AgentCount]
    stages_reached: list[AgentCount]
    last_run_at: datetime | None
    unmapped_identity_count: int
    data_mode: str = "ingested"
    reference_time: datetime
    note: str = "Ingested agent result. Advisory only. Agent audit logs do not replace operator history."
